mev-boost-rs/src/service.rs

```diff
@@ -30,18 +30,19 @@ impl Default for Config {
     }
 }
 
-fn parse_url(input: &str) -> Option<Url> {
-    if input.is_empty() {
-        None
-    } else {
-        input
-            .parse()
-            .map_err(|err| {
-                tracing::warn!("error parsing relay from URL: `{err}`");
-                err
-            })
-            .ok()
+fn parse_relay_endpoints(relay_urls: &[String]) -> Vec<RelayEndpoint> {
+    let mut relays = Vec::new();
+
+    for relay_url in relay_urls {
+        match relay_url.parse::<Url>() {
+            Ok(url) => match RelayEndpoint::try_from(url.clone()) {
```

Comment:
should be able to skip the clone here right?

```suggestion
            Ok(url) => match RelayEndpoint::try_from(url) {
```

Reply:
No the compiler complains about this

Reply:
what does it complain about? this seems to be fine: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f36816a9416a02df4132432339f555bc